packages/errors/scripts/extract-errors.ts

```diff
@@ -0,0 +1,189 @@
+import * as fs from 'fs';
+import { compile } from 'handlebars';
+import * as path from 'path';
+import ts from 'typescript';
+import * as util from 'util';
+import * as ux from './ux';
+
+type PackageError = { code: string, documentation: string };
+type PackageErrors = { package: string, errors: PackageError[] };
+
+const MONGOSH_ERRORS_DOC_TAG = 'mongoshErrors';
+
+(async function() {
+  const pathToPackages = path.resolve(process.argv[process.argv.length - 1]);
+  if (!pathToPackages || !(await isDirectory(pathToPackages))) {
+    ux.fatal('Could not find given packages directory:', pathToPackages);
+    return;
+  }
+
+  const packageErrors: PackageErrors[] = [];
+
+  const packages = await collectPackages(pathToPackages);
+  for await (const packagePath of packages) {
+    const dirName = path.basename(packagePath);
+    ux.info(`Processing ${dirName}...`);
+    const pe = await processPackage(packagePath);
+    if (pe) {
+      packageErrors.push(pe);
+      ux.note(`Found ${pe.errors.length} mongosh error enums.\n`);
+    } else {
+      ux.quiet('No mongosh error enums found.\n');
+    }
+  }
+
+  await renderErrorOverview(packageErrors);
+
+  ux.success('👏👏👏👏');
+  ux.success('Wrote generated overview page to: error-overview.md');
+  ux.success('👏👏👏👏');
+})();
```

Comment:
```suggestion
})().catch(err => process.nextTick(() => { throw err; }));
```

Otherwise errors will not lead to an exit with a non-zero exit code (yet, that will happen in a future version of Node.js)